examples/bri-3/src/bri/identity/bpiSubjects/agents/bpiSubjects.agent.ts

```diff
@@ -1,22 +1,58 @@
-import { BadRequestException, Injectable } from "@nestjs/common";
+import { BadRequestException, Injectable, NotFoundException } from "@nestjs/common";
 import { BpiSubject } from "../models/bpiSubject";
 import { BpiSubjectType } from "../models/bpiSubjectType.enum";
 
 import { uuid } from 'uuidv4'; 
+import { BpiSubjectRepository } from "../persistence/bpiSubjects.repository";
+import { NotFoundError } from "rxjs";
  
+// Agent methods have extremely declarative names and perform a single task 
 @Injectable()
 export class BpiSubjectAgent {
-  // Agent methods have extremely declarative names and perform a single task 
+
+  constructor(private repo: BpiSubjectRepository) { }
   public throwIfCreateBpiSubjectInputInvalid(name: string, desc: string, pk: string) {
     // This is just an example, these fields will be validated on the DTO validation layer
     // This validation would check internal business rules (i.e. bpiSubject must have public key in the format defined by the participants..) 
     if (!name) {
-      // We stop execution in case of critical errors by throwing a simple exception
       throw new BadRequestException("Name cannot be empty.")
     };
   }
 
   public createNewExternalBpiSubject(name :string, description: string, publicKey: string): BpiSubject {
     return new BpiSubject(uuid(), name, description, BpiSubjectType.External, publicKey );
   }
+
+  public async fetchUpdateCandidateAndThrowIfUpdateValidationFails(id: string, name: string, desc: string, pk: string): Promise<BpiSubject> {
+
+    const bpiSubjectToUpdate = await this.repo.getBpiSubjectById(id);
+
+    if(!bpiSubjectToUpdate) {
+      throw new NotFoundException(`Bpi Subject with id: ${id} does not exist.`)
+    }
+
+    // Example additional checks
+    if (!name) {
```

Comment:
couple nitpicks here:
- check if name is not defined before checking db seems cleaner
- empty lines on 27 and 49
- unclear comment on 34, not sure what example aditional check means
- i think desc and pk were also mandatory

Reply:
Removed the check completely. It is just an example but it is confusing. These input fields will be checked by the input DTO validation, once we implement it. In the agent, we ll add additional validation of the business rules, once we get to them.